shell/platform/fuchsia/flutter/platform_view_unittest.cc

```diff
@@ -0,0 +1,202 @@
+  // |flutter::PlatformView::Delegate|
+  void OnPlatformViewSetAccessibilityFeatures(int32_t flags) {
+    semantics_features_ = flags;
+  }
+  // |flutter::PlatformView::Delegate|
+  void OnPlatformViewRegisterTexture(
+      std::shared_ptr<flutter::Texture> texture) {}
+  // |flutter::PlatformView::Delegate|
+  void OnPlatformViewUnregisterTexture(int64_t texture_id) {}
+  // |flutter::PlatformView::Delegate|
+  void OnPlatformViewMarkTextureFrameAvailable(int64_t texture_id) {}
+
+  bool SemanticsEnabled() const { return semantics_enabled_; }
+  int32_t SemanticsFeatures() const { return semantics_features_; }
+
+ private:
+  bool semantics_enabled_ = false;
+  int32_t semantics_features_ = 0;
+};
+
+TEST_F(PlatformViewTests, ChangesAccessibilitySettings) {
+  sys::testing::ServiceDirectoryProvider services_provider(dispatcher());
+
+  MockPlatformViewDelegate delegate;
+  zx::eventpair a, b;
+  zx::eventpair::create(/* flags */ 0u, &a, &b);
+  auto view_ref = fuchsia::ui::views::ViewRef({
+      .reference = std::move(a),
+  });
+  auto view_ref_control = fuchsia::ui::views::ViewRefControl({
+      .reference = std::move(b),
+  });
+  flutter::TaskRunners task_runners =
+      flutter::TaskRunners("test_runners", nullptr, nullptr, nullptr, nullptr);
+
+  EXPECT_FALSE(delegate.SemanticsEnabled());
+  EXPECT_EQ(delegate.SemanticsFeatures(), 0);
+
+  auto platform_view = flutter_runner::PlatformView(
+      delegate,                               // delegate
+      "test_platform_view",                   // label
+      std::move(view_ref_control),            // view_ref_control
+      std::move(view_ref),                    // view_ref
+      std::move(task_runners),                // task_runners
+      services_provider.service_directory(),  // runner_services
+      nullptr,  // parent_environment_service_provider_handle
+      nullptr,  // session_listener_request
+      nullptr,  // on_session_listener_error_callback
+      nullptr,  // session_metrics_did_change_callback
+      nullptr,  // session_size_change_hint_callback
+      nullptr,  // on_enable_wireframe_callback,
+      0u        // vsync_event_handle
+  );
+
+  RunLoopUntilIdle();
+
+  platform_view.SetSemanticsEnabled(true);
+
+  EXPECT_TRUE(delegate.SemanticsEnabled());
+  EXPECT_EQ(delegate.SemanticsFeatures(),
+            static_cast<int32_t>(
+                flutter::AccessibilityFeatureFlag::kAccessibleNavigation));
+
+  platform_view.SetSemanticsEnabled(false);
+
+  EXPECT_FALSE(delegate.SemanticsEnabled());
+  EXPECT_EQ(delegate.SemanticsFeatures(), 0);
+}
+
+// Test to make sure that PlatformView correctly registers messages sent on
+// the "flutter/platform_views" channel, correctly parses the JSON it receives
+// and calls the EnableWireframeCallback with the appropriate args.
+TEST_F(PlatformViewTests, EnableWireframeTest) {
+  sys::testing::ServiceDirectoryProvider services_provider(dispatcher());
+  MockPlatformViewDelegate delegate;
+  zx::eventpair a, b;
```

Comment:
Why do we need this?
SetUp() is already creating eventpair using viewrefs
This looks unnecessary

Reply:
That's for a different test... the a11y tests are structured differently.

Reply:
Sorry I mixed them. Not very used to this ui.

Reply:
[Optional] I would recommend changing names of a and b here to something more meaningful.